src/transformers/models/t5/tokenization_t5.py

```diff
@@ -301,14 +301,15 @@ def __setstate__(self, d):
         self.sp_model.Load(self.vocab_file)
 
     def tokenize(self, text: "TextInput", **kwargs) -> List[str]:
-        if not text.startswith(" "):
-            text = " " + text
+        text = SPIECE_UNDERLINE + text
         return super().tokenize(text, **kwargs)
 
     def _tokenize(self, text: str) -> List[str]:
         """Take as input a string and return a list of strings (tokens) for words/sub-words"""
+        if text.startswith(SPIECE_UNDERLINE):
+            return self.sp_model.encode(text[1:], out_type=str)
         tokens = self.sp_model.encode(text, out_type=str)
-        if not text.startswith(" ") and tokens[0] == SPIECE_UNDERLINE:
+        if not text.startswith((SPIECE_UNDERLINE)) and tokens[0] == SPIECE_UNDERLINE:
```

Comment:
We don't need to check `text.startswith(SPIECE_UNDERLINE)` again, as there's an early return on L310, so we know all text here doesn't start with `SPIECE_UNDERLINE`

```suggestion
        if tokens[0] == SPIECE_UNDERLINE:
```

Reply:
Indeed!